### PR TITLE
Support for nocase flag in non-glob string pattern

### PR DESCRIPTION
**Pull request by idobh2**
Noticed it wasn't supported for non-globs patterns - added a fix and tests

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -44,7 +44,11 @@ utils.filename = function filename(fp) {
 
 utils.isPath = function isPath(pattern, opts) {
   return function(fp) {
-    return pattern === utils.unixify(fp, opts);
```
Comment (jonschlinkert) [inline lib/utils.js]: here, we'll need to add `if (!opts) opts = {}` to ensure the object exists. the tests are currently failing on travis, I think this should fix it.

also, great idea on the check, and thank you!

**Comment by idobh2** [comment]
Matched it to other default _opts_ assigning in the code, hope it's ok :)

**Comment by jonschlinkert** [comment]
Sorry for the delay. I'll try to get this merged in tomorrow! thanks!

**Comment by phated** [comment]
Ref https://github.com/js-cli/js-liftoff/issues/72 maybe?

**Comment by jonschlinkert** [comment]
published to npm and released as 2.3.9. thanks @idobh2! and sorry for the delay. 

@phated I'll also follow up on liftoff and gulp

**Comment by phated** [comment]
2.3.9 doesn't look correct, did you publish with node 6 😉

**Comment by phated** [comment]
@jonschlinkert after poking at this a bit, it looks like https://github.com/jonschlinkert/micromatch/pull/65 was published as the 2.3.9 instead of this change.  Could that have happened somehow?

**Comment by jonschlinkert** [comment]
> Could that have happened somehow?

yes! I just realized that myself and was hoping no one else did lol. give me a minute, my connection keeps going down, I'll get it fixed

**Comment by jonschlinkert** [comment]
this is what happens when you rush. sorry I had 2.3.9 on my brain when I pushed up, should be fixed now.

**Comment by phated** [comment]
@jonschlinkert thanks.  Contacting out-of-band for a followup (findup-sync related)

**Comment by jonschlinkert** [comment]
no prob. let me know if the latest patch works for you. If not and the issue is in [this code in liftoff](https://github.com/js-cli/node-findup-sync/blob/master/index.js#L74-L91) I'll need to look into it more. it looks like the only code that does any comparison in that function is `exists`, which uses node's `fs.stat` I believe.

**Comment by phated** [comment]
@jonschlinkert yeah, that chunk of code looks to be the problem.  Wondering if I should rollback until we can work through it.

**Comment by jonschlinkert** [comment]
Hmm, strange. I thought `fs.stat` wasn't case sensitive. I'll pull it down and look at it now. feel free to roll back, or I can give you a progress update in 5-10 min and you can decide then. totally your call

**Comment by jonschlinkert** [comment]
cool I see you rolled back. that gives me a little more time. I'll let you know when I find something.

**Comment by phated** [comment]
yeah, i wanted to roll back so we can get a proper fix with regression coverage, etc.

**Comment by jonschlinkert** [comment]
that's exactly what I was about to say. we just ran it on linux and discovered the same. still working on it

**Pull request merged**